src/App.ts

```diff
@@ -369,101 +382,45 @@ export default class App {
     }
 
     /* --------------------- Initialize receiver ---------------------- */
-    if (receiver !== undefined) {
-      // Custom receiver supplied
-      if (this.socketMode === true) {
-        // socketMode = true should result in SocketModeReceiver being used as receiver
-        // TODO: Add case for when socketMode = true and receiver = SocketModeReceiver
-        // as this should not result in an error
-        throw new AppInitializationError('You cannot supply a custom receiver when socketMode is set to true.');
-      }
-      this.receiver = receiver;
-    } else if (this.socketMode === true) {
-      if (appToken === undefined) {
-        throw new AppInitializationError('You must provide an appToken when socketMode is set to true. To generate an appToken see: https://api.slack.com/apis/connections/socket#token');
-      }
-      this.logger.debug('Initializing SocketModeReceiver');
-      this.receiver = new SocketModeReceiver({
-        appToken,
-        clientId,
-        clientSecret,
-        stateSecret,
-        redirectUri,
-        installationStore,
-        scopes,
-        logger,
-        logLevel: this.logLevel,
-        installerOptions: this.installerOptions,
-        customRoutes,
-      });
-    } else if (signatureVerification === true && signingSecret === undefined) {
-      // Using default receiver HTTPReceiver, signature verification enabled, missing signingSecret
-      throw new AppInitializationError(
-        'signingSecret is required to initialize the default receiver. Set signingSecret or use a ' +
-          'custom receiver. You can find your Signing Secret in your Slack App Settings.',
-      );
-    } else {
-      this.logger.debug('Initializing HTTPReceiver');
-      this.receiver = new HTTPReceiver({
-        signingSecret: signingSecret || '',
-        endpoints,
-        port,
-        customRoutes,
-        processBeforeResponse,
-        signatureVerification,
-        clientId,
-        clientSecret,
-        stateSecret,
-        redirectUri,
-        installationStore,
-        scopes,
-        logger,
-        logLevel: this.logLevel,
-        installerOptions: this.installerOptions,
-      });
-    }
+    this.receiver = this.initReceiver(
```

Comment:
This is so nice! 💯

Reply:
Agreed, I like splitting up the initialization into discrete functions!